AEPCampaignClassic/Tests/UnitTests/CampaignClassicPublicAPITests.swift

```diff
@@ -0,0 +1,136 @@
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+@testable import AEPCore
+@testable import AEPCampaignClassic
+import XCTest
+
+class CampaignClassicPublicAPITests: XCTestCase {
+    
+    let SAMPLE_PUSHTOKEN_DATA = "PushToken".data(using: .utf8)!
+    let SAMPLE_PUSHTOKEN_AS_HEXSTRING = "50757368546F6B656E"
+    let SAMPLE_INFO : [String : String] = ["key" : "value"]
+    let SAMPLE_ADDITIONAL_DATA = ["additionalDataKey" : "additionalDataValue"]
+
+    override func setUpWithError() throws {
+        EventHub.shared.start()
+        registerMockExtension(MockExtension.self)
+    }
+
+    override func tearDownWithError() throws {
+        unregisterMockExtension(MockExtension.self)
+    }
+
+    func test_registerDevice() throws {
+        let SAMPLE_USER_KEY = "UserKey"
+        let expectation = XCTestExpectation(description: "Register Device API should dispatch appropriate event")
+        
+        // event dispatch verification
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(type: EventType.campaign,
+                                                                                    source: EventSource.requestContent) { event in
+            // unwrap optionals
+            let isRegisterDeviceEvent = try? XCTUnwrap(event.data?[CampaignClassicConstants.EventDataKeys.CampaignClassic.REGISTER_DEVICE] as? Bool)
+            let deviceToken = try? XCTUnwrap(event.data?[CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_TOKEN] as? String)
+            let userKey = try? XCTUnwrap(event.data?[CampaignClassicConstants.EventDataKeys.CampaignClassic.USER_KEY] as? String)
+            let additionalParameters = try? XCTUnwrap(event.data?[CampaignClassicConstants.EventDataKeys.CampaignClassic.ADDITIONAL_PARAMETERS] as? [String : String])
+            
+            // verify
+            XCTAssertEqual(isRegisterDeviceEvent, true)
+            XCTAssertEqual(deviceToken, self.SAMPLE_PUSHTOKEN_AS_HEXSTRING)
+            XCTAssertEqual(userKey, SAMPLE_USER_KEY)
+            XCTAssertEqual(additionalParameters, self.SAMPLE_ADDITIONAL_DATA)
+            
+            // verify deviceInfo
+            let deviceInfo = try? XCTUnwrap(event.data?[CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO] as? [String : String])
+            XCTAssertNotNil(deviceInfo?[CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO_KEY_DEVICE_MODEL])
```

Comment:
this and the next one are two different checks.. this one checks for the `device model` and the next one for `device name`
The reason why I do only nil check for `deviceModel` is because it might be different when I run on CI